app/ruby/controller.py

```diff
@@ -47,4 +50,76 @@ def get_challenge(self, id):
 
     def get_all_challenges(self):
         challenges = self.dao.get_challenges_data()
-        return jsonify({'challenges': challenges})
\ No newline at end of file
+        return jsonify({'challenges': challenges})
+
+    def modify_challenge(self, id, code_file, tests_code_file, json):
+        data = {'repair_objective': None, 'complexity': None}
+        data.update(json['challenge'])
+        challenge = self.dao.get_challenge(id)
+        del challenge['best_score']
+        old_challenge = RubyChallenge(**challenge)
+        new_challenge = RubyChallenge(data['repair_objective'], data['complexity'])
+        ruby_tmp = gettempdir() + '/ruby-tmp/'
+        mkdir(ruby_tmp)
+        #If files names are in the request, set new_code names to them. If not, take old_challenge name.
+        nc_code_name = data['source_code_file_name'] if 'source_code_file_name' in data else old_challenge.code.get_file_name()
+        nc_test_name = data['test_suite_file_name'] if 'test_suite_file_name' in data else old_challenge.tests_code.get_file_name()
+        
+        if code_file is not None:
+            new_challenge.set_code(ruby_tmp, nc_code_name, code_file)
+            if not new_challenge.save_code():
+                rmtree(ruby_tmp)
+                return make_response(jsonify({'code': 'couldnt save code'}), 400)
+            if not new_challenge.code_compile():
+                rmtree(ruby_tmp)
+                return make_response(jsonify({'code': 'code doesnt compile'}), 400)
+        else: #If no file is passed, set the old_challenge code as the new one (Needed to check dependencies)
+            old_challenge.copy_code(ruby_tmp)
+            new_challenge.set_code(ruby_tmp, old_challenge.code.get_file_name(), None)
```

Comment:
Maybe the last parameter None could be a default value for that parameter in the declaration of the method